src/myframe.h

```diff
@@ -140,9 +140,17 @@ struct MyFrame : wxFrame {
             // blit it over via a MemoryDC?
         #endif
         #ifdef __WXGTK__
-            // Currently on Linux csf is always 1, so this is useless.
-            // FIXME: On a high-DPI display we get low res images even though the display is
-            // capable of better!
+            // On X11 platform, there is no scaling factor beside 1.0, but the DPI can be changed.
+
+            // On Wayland platform, there might be a scaling factor beside 1.0, 
+            // but the DPI is assumed to be a fixed value by default.
+
+            // If upscaling is active in Wayland compositors, the reported bitmap to paint on is downsized (and vice versa).
+            // The scaling is handled by the GTK toolkit and the Wayland compositor. 
+            // This is why csf and csf_orig should be set to 1.0 to let scaling be handled by the
+            // toolkit and compositor.
+            csf_orig = 1.0;

```

Comment:
Why change `csf_orig` ? the whole point of that variable is that it is the original value returned by `GetContentScaleFactor`.

So on Wayland, does `GetContentScaleFactor` return other values than 1.0?

Reply:
Thanks for looking into this! I removed `csf_orig`. Yes, under Wayland compositors and scaling active, the active scaling factor is reported.